File: frontend/src/app/cad/infrastructure/abrir-ocorrencia.request.ts
export interface AbrirOcorrenciaRequest {
  natureza: string;
  descricao: string;
  gravidade: string;
  endereco: string;
  latitude: number;
  longitude: number;
  protocolo: string;
  telefone?: string | null;
  pabxUid?: string | null;
  pontoReferencia?: string | null;
}

File: backend/infrastructure/src/main/java/br/gov/mt/sesp/cicc/infrastructure/rest/AbrirOcorrenciaRequest.java
package br.gov.mt.sesp.cicc.infrastructure.rest;

import br.gov.mt.sesp.cicc.application.cad.AbrirOcorrenciaUseCase;

public record AbrirOcorrenciaRequest(
        String natureza,
        String descricao,
        String gravidade,
        String endereco,
        double latitude,
        double longitude,
        String protocolo,
        String telefone,
        String pabxUid,
        String pontoReferencia
) {

    AbrirOcorrenciaUseCase.Input toInput() {
        return new AbrirOcorrenciaUseCase.Input(
                natureza,
                descricao,
                gravidade,
                endereco,
                latitude,
                longitude,
                protocolo,
                telefone,
                pabxUid,
                pontoReferencia
        );
    }
}
